### PR TITLE
Add testing

### PR DESCRIPTION
**Pull request by praveenperera**
Found #8 while adding this

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1252,12 +1252,19 @@
...
+        "@types/chai": "^4.2.5",
+        "@types/lodash": "^4.14.149",
+        "@types/mocha": "^5.2.7",
         "@types/node": "^6.0.40",
+        "chai": "^4.2.0",
+        "lodash": "^4.17.15",
+        "mocha": "^6.2.2",
+        "ts-node": "^8.5.2",
         "tslint": "^5.16.0",
-        "typescript": "^3.5.1",
```
Comment (joshmanders) [inline package.json]: TypeScript is still needed, isn't it?

**Comment by praveenperera** [comment]
Oops good catch @joshmanders, fixed

**Comment by heybourn** [comment]
Does this still look okay with the latest merge?

**Comment by praveenperera** [comment]
I rebase on master its all good still.

@heybourn do you have a preference between keeping the tests in `src/`, or moving them to a top level `/tests` folder?

Right now I've keeping them together with the files in `src`.

**Comment by heybourn** [comment]
I'm not too worried. What is best practice usually?

**Comment by praveenperera** [comment]
I am not either we can leave it like this for now and move it later if we need to for whatever reason.

Re best practices I've seen it both ways, @joshmanders any input on this?

**Comment by joshmanders** [comment]
Depends on your preferences really. Large projects can benefit from keeping the tests with the source, so you're not pile driving through trees of directories mirroring, but for this project it might be better to have them in a top level `tests/` folder.

Also I'd probably swap to using [Jest](https://jestjs.io) over Mocha as it's more user friendly and has basically become the goto test harness for JavaScript projects.

**Comment by joshmanders** [comment]
Switching to Jest can be pretty easy, just drop `mocha` and `chai` from `devDependencies`, add `jest` and remove all `import {expect} from 'chai'` calls, as `expect` is already apart of Jest's global functions (like `describe` and `it`)

**Comment by heybourn** [comment]
@praveenperera let's roll with what Josh has said here. I like the idea of a /tests folder for now.

**Comment by praveenperera** [comment]
Done we now have the new scripts:

`npm run test`
`npm run test:watch`
`npm run test:coverage`

**Comment by joshmanders** [comment]
LGTM 👍

**Comment by heybourn** [comment]
`File '/Users/ryan/Desktop/github/headwind/tests/utils.spec.ts' is not under 'rootDir' '/Users/ryan/Desktop/github/headwind/src'. 'rootDir' is expected to contain all source files.`

**Pull request merged**